### PR TITLE
`gdock` update

### PR DESCRIPTION
**Pull request by rvhonorato**
This PR just updates the third-party module `gdock` to the latest code.

### CODE REVIEW

File: src/haddock/modules/sampling/gdock/__init__.py [hunk 1]
```diff
@@ -114,19 +111,32 @@ def _run(self):
...
 
         # retrieve the structures
         output_structures = []
-        structure_folder = Path(self.path, 'gdock-integration/structures')
+        structure_folder = Path('gdock-integration/structures')
         for model in structure_folder.glob('*pdb'):
-            pdb = PDBFile(model, path=model.parent)
+            # Make sure the output is tidy, this should be handled
+            #  by gdock, but check it here to be sure
+            tmp_f = tempfile.NamedTemporaryFile(delete=False)
```
Comment (joaomcteixeira) [inline src/haddock/modules/sampling/gdock/__init__.py:134]: I think is best if you write this with the following:

```python
from pdbtools import pdb_tidy 
with open(model, 'r') as fin:
    lines = list(pdb_tidy.run(fin, strick=False))  # be explicit in the `strict`

with open(model, 'w') as fout:
    fout.write(os.linesep.join(lines))
```
Don't know if the `os.linesep` is needed. With this, you avoid creating a temporary file. I think after #144 this part is no longer needed. I will have that in mind.

Aside from this, :rocket:

File: src/haddock/modules/sampling/gdock/__init__.py [hunk 2]
```diff
@@ -114,19 +111,32 @@ def _run(self):
...
-            cmd = f'{sys.executable} {gdock_exec} --dry input.toml'
+        cmd = f'{sys.executable} {gdock_exec} --dry input.toml'
 
-            subprocess.call(cmd, shell=True)
+        subprocess.call(cmd, shell=True)
 
         # retrieve the structures
         output_structures = []
-        structure_folder = Path(self.path, 'gdock-integration/structures')
+        structure_folder = Path('gdock-integration/structures')
```
Comment (joaomcteixeira) [inline src/haddock/modules/sampling/gdock/__init__.py:130]: Currently, `structure_folder` will be searched inside the `0*_gdock` folder in the `run_dir`. Make sure this is what you want, or if you need some `../data` first.

File: src/haddock/modules/sampling/gdock/__init__.py (hunk 1 above)
Comment (rvhonorato) [reply src/haddock/modules/sampling/gdock/__init__.py:134]: thanks! I keep trying to find reasons to use tempfile but there is never a good one 😢

File: src/haddock/modules/sampling/gdock/__init__.py (hunk 2 above)
Comment (rvhonorato) [reply src/haddock/modules/sampling/gdock/__init__.py:130]: yep took me a while to figure this out, as its now the next module can find the structures fine.

File: src/haddock/modules/sampling/gdock/__init__.py (hunk 1 above)
Comment (joaomcteixeira) [reply src/haddock/modules/sampling/gdock/__init__.py]: Yeah, we are always trying to reduce the disk usage, so... :stuck_out_tongue_closed_eyes:

File: src/haddock/modules/sampling/gdock/__init__.py (hunk 2 above)
Comment (joaomcteixeira) [reply src/haddock/modules/sampling/gdock/__init__.py:130]: After #219 each module runs on its own folder. We need to keep this in mind now when defining paths.

**Pull request merged**